bigquery_etl/glam/templates/probe_counts_v1.sql

```diff
@@ -1,8 +1,48 @@
 {{ header }}
-{% if is_scalar %}
-    {% include "probe_counts_v1.udf.scalar.sql" %}
-{% else %}
-    {% include "probe_counts_v1.udf.histogram.sql" %}
+{% if not is_scalar %}
+
+CREATE TEMP FUNCTION udf_get_buckets(
+  metric_type STRING,
+  range_min INT64,
+  range_max INT64,
+  bucket_count INT64
+)
+RETURNS ARRAY<INT64> AS (
+  (
+    WITH buckets AS (
+      SELECT
+        CASE
+        WHEN
+          metric_type = 'timing_distribution'
+        THEN
+          -- https://mozilla.github.io/glean/book/user/metrics/timing_distribution.html
+          mozfun.glam.histogram_generate_functional_buckets(2, 8, range_max)
+        WHEN
+          metric_type = 'memory_distribution'
+        THEN
+          -- https://mozilla.github.io/glean/book/user/metrics/memory_distribution.html
+          mozfun.glam.histogram_generate_functional_buckets(2, 16, range_max)
+        WHEN
+          metric_type = 'custom_distribution_exponential'
+        THEN
+          mozfun.glam.histogram_generate_exponential_buckets(range_min, range_max, bucket_count)
+        WHEN
+          metric_type = 'custom_distribution_linear'
+        THEN
+          mozfun.glam.histogram_generate_linear_buckets(range_min, range_max, bucket_count)
+        ELSE
+          []
```

Comment:
I know this isn't a new change but I'm curious if this is an explicitly allowed behaviour: passing an unknown metric type.  What's the behaviour if no buckets are returned?

Reply:
This would result in an empty list from this udf. If the buckets are unnested, it would cause the row to disappear from the resulting set. Otherwise, the histogram would probably just be empty.

```sql
select array_agg(cast(item as int64)) from unnest([]) item
-- {"_f0": []|}
```